### PR TITLE
HW6: create server with Express

### PR DESCRIPTION
**Pull request by YuliaBarbashova**

### CODE REVIEW

File: HW6/controllers/auth.controller.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  async (req, res, next) => {
+    try {
+      const user = (await UserService.createNewUser(
+        req.body.email,
+        req.body.password,
+        req.body.role
+      )) as Partial<UserEntity>;
+
+      delete user.password;
+      return res.status(200).json({ data: user, error: null });
```
Comment (AlexeyBarkow) [inline HW6/controllers/auth.controller.ts]: 201 status would be better

File: HW6/controllers/cart.controller.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import { CartEntity, OrderEntity } from "../types/index.ts";
+import { validateBody, cartSchema } from "../validators/index.ts";
+
+const CartController = Router();
+
+CartController.get("/", authMiddleware, async (req, res, next) => {
+  const userId = req.headers["x-user-id"] as string;
+
+  try {
+    const cart: CartEntity | undefined = await CartService.getUserCart(userId);
```
Comment (AlexeyBarkow) [inline HW6/controllers/cart.controller.ts]: According to requirements, this endpoint should create card if it's missing: `get user cart or create if missing endpoint is implemented based on Swagger`

File: HW6/controllers/cart.controller.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+      });
+    }
+
+    return res.status(200).json({
+      data: { success: true },
+      error: null,
+    });
+  } catch (err) {
+    if ((err as Error).message === "Cart not found") {
+      return res.status(404).json({
```
Comment (AlexeyBarkow) [inline HW6/controllers/cart.controller.ts:86]: Looks like one of 404 conditions is redundant

File: HW6/repositories/order.repository.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+export const createNewOrder = (
+  cart: CartEntity,
+  payment?: PaymentEntity,
+  delivery?: DeliveryEntity,
+  comments?: string
+) => {
+  return OrderModel.addNewOrder({
+    userId: cart.userId,
+    cartId: cart.id,
+    items: cart.items,
```
Comment (AlexeyBarkow) [inline HW6/repositories/order.repository.ts:13]: looks like you forgot to create a copy of items: when order is created, it has an empty items array

File: HW6/validators/body.schemas.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import Joi from "joi";
+
+export const cartSchema = Joi.object({
+  productId: Joi.string().required(),
+  count: Joi.number().integer().min(0).required(),
```
Comment (AlexeyBarkow) [inline HW6/validators/body.schemas.ts]: count with string value (i.e `"12"`) passes this validation

**Review by AlexeyBarkow — CHANGES_REQUESTED** [review]

File: HW6/repositories/order.repository.ts (hunk 4 above)
Comment (YuliaBarbashova) [reply HW6/repositories/order.repository.ts:13]: Perhaps I didn't understand correctly, but the list of items is saved in the order. I simply clear the list of products before deleting the card. 
image
image
image

File: HW6/validators/body.schemas.ts (hunk 5 above)
Comment (YuliaBarbashova) [reply HW6/validators/body.schemas.ts]: Updated

File: HW6/controllers/cart.controller.ts (hunk 3 above)
Comment (YuliaBarbashova) [reply HW6/controllers/cart.controller.ts:86]: Updated

File: HW6/controllers/cart.controller.ts (hunk 2 above)
Comment (YuliaBarbashova) [reply HW6/controllers/cart.controller.ts]: Oops, fixed

File: HW6/controllers/auth.controller.ts (hunk 1 above)
Comment (YuliaBarbashova) [reply HW6/controllers/auth.controller.ts]: Updated

**Pull request merged**